### PR TITLE
Port Servers of Happiness code to Python3.

### PR DESCRIPTION
**Pull request by ccxcz**
This change is [Reviewable](https://reviewable.io/reviews/tahoe-lafs/tahoe-lafs/689)

### CODE REVIEW

File: src/allmydata/immutable/happiness_upload.py [hunk 1]
```diff
@@ -1,5 +1,11 @@
 
-from Queue import PriorityQueue
+import itertools
+try:
+    from queue import PriorityQueue
+except ImportError:
+    from Queue import PriorityQueue
```
Comment (exarkun) [inline src/allmydata/immutable/happiness_upload.py]: `six.moves.queue`?

File: src/allmydata/immutable/happiness_upload.py [hunk 2]
```diff
@@ -260,9 +266,9 @@ def _servermap_flow_graph(peers, shares, servermap):
     #print "share_to_index %s" % share_to_index
     #print "servermap %s" % servermap
     for peer in peers:
-        if servermap.has_key(peer):
+        if peer in servermap:
```
Comment (exarkun) [inline src/allmydata/immutable/happiness_upload.py]: Can you document the `servermap` parameter as a `dict`?

**Comment by exarkun** [comment]
Thanks.  I think there is still a parameter in some changed could that could be documented - `servermap` accepted by `_servermap_flow_graph`.  Also, the lint job will fail until there is a news fragment - which requires a ticket in trac describing the goals of the change (I am a little unclear on them, which is why tickets are required :smile: ).  Thanks again.

**Comment by itamarst** [comment]
This has been superseded, the module has been ported now.

**Pull request closed without merging**